[TSK-111] Feat: Add Login page logic and area searching page

glenn-syj opened the pull request:
- 로그인 로직 전 범위 구현
- 지역 검색 페이지 내 로직 구현
- 회원가입 로직 전 범위 구현

glenn-syj committed:
[TSK-102] Design: 로그인 화면 UI 수정

로그인 화면 UI를 수정했습니다.
- 임시 로고 추가
- Column 일렬 배열
- 카카오 디자인 가이드 준수

glenn-syj committed:
[TSK-111] Feat: 지역 설정 화면 검색 기본 컴포넌트 및 API 연동

- API 연동 결과는 다시 한 번 확인 필요 (merge X!)
- 화면 및 로직 검토 필요
- 추후 연동 완료 후 RepositoryImpl 을 이용해 viewModel과 repository 분리 필요

glenn-syj committed:
[TSK-111] 지역 검색 및 기본 페이지 구현

- 지역 선택 페이지 기본 화면 구현

- 지역 선택 API 연동

- 앞으로: 지역 선택 시 표기, 정보 취합해 SignUp 할 때 받아온 정보와 지역 모두 파라미터로 전달

gazi52 committed:
[TSK-99] 에러 파싱

ErrorBody의 message항목 분리
 - UI 디자인 작업 시 message 내용에 따라 조건 미충족 항목에 표시 필요

gazi52 committed:
Merge branch 'login-page' of https://github.com/wafflestudio21-5/team2-android into login-page

glenn-syj committed:
[TSK-111] Feat: Add Login page logic and area searching page

glenn-syj committed:
Merge branch 'main' into feature-loginLogic

glenn-syj committed:
[TSK-111] Fix: viewModel 관련 오류 해결 중

gazi52 reviewed (approved):
디자인부터 카카오 로그인까지... 정말 고생 많으셨습니다!

glenn-syj committed:
[TSK-129] Fix: 컨플릭트 해결 후 발생 로그인 관련 에러 해결

- 로그인 및 회원가입 가능
- baseURI 수정
- 무한스크롤 논의 필요
- viewModel 내 변수들 stateFlow 화

Resolves: TSK-111
